docs: Style and formatting improvements to User Guide

olivergondza opened the pull request:
<!--
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.
-->

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [n/a] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [n/a] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [n/a] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [n/a] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [n/a] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [n/a] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

<!-- Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request. -->

rumstead commented on docs/user-guide/auto_sync.md:
```diff
@@ -27,7 +27,7 @@ spec:
 
 !!!note 
-  Setting `spec.syncPolicy.automated.enabled` flag to null will be treated as automated sync as enabled. When using `enabled` field set to false, fields like `prune`, `self-heal` and `allowEmpty` can be set without enabling them.
+    Setting `spec.syncPolicy.automated.enabled` flag to null will be treated as automated sync is enabled. When using `enabled` field set to false, fields like `prune`, `selfHeal` and `allowEmpty` can be set without enabling them.
```
while you are there :)

```suggestion
    Setting the `spec.syncPolicy.automated.enabled` flag to null will be treated as if automated sync is enabled. When the `enabled` field is set to false, fields like `prune`, `selfHeal` and `allowEmpty` can be set without enabling them.
```

rumstead commented on docs/user-guide/auto_sync.md:
```diff
@@ -100,7 +100,7 @@ Disabling self-heal does not guarantee that live cluster changes won't be revert
   application parameters. If the most recent successful sync in the history was already performed
   against the same commit-SHA and parameters, a second sync will not be attempted, unless `selfHeal` flag is set to true.
-* If `selfHeal` flag is set to true then sync will be attempted again after self heal timeout (5 seconds by default)
+* If `selfHeal` flag is set to true then sync will be attempted again after self-heal timeout (5 seconds by default)
```
```suggestion
* If the `selfHeal` flag is set to true, then the sync will be attempted again after self-heal timeout (5 seconds by default)
```

rumstead commented on docs/user-guide/diff-strategies.md:
```diff
@@ -17,16 +17,14 @@ Argo CD currently has 3 different strategies to calculate diffs:
 
-!!! warning "Beta Feature (Since v2.5.0)"
-    This feature is in the [Beta][1] stage. It is generally considered stable, but there may be unhandled edge cases.
+!!! warning "Feature Discontinued"
```
Is this true?

olivergondza replied:
It was the last sentence of the paragraph. I have just moved it to a more prominent place, as IMO it makes no sense to warn the feature is in beta _and_ that it will be removed. In the light of the latter, the former is insignificant.

rumstead commented on docs/user-guide/multiple_sources.md:
```diff
@@ -80,8 +80,7 @@ If the `path` field is set in the `$values` source, Argo CD will attempt to gene
 !!! note
-    Sources with the `ref` field set must not also specify the `chart` field. Argo CD does not currently support using  
-    another Helm chart as a source for value files.
+    Sources with the `ref` field set must not also specify the `chart` field. Argo CD does not currently support using another Helm chart as a source for value files.
```
```suggestion
    Sources with the `ref` field set cannot include the `chart` field. Currently, Argo CD does not support using another Helm chart as a source for value files.
```

rumstead commented on docs/user-guide/parameters.md:
```diff
@@ -26,8 +26,8 @@ argocd app set guestbook -p ingress.enabled=true
-The `argocd app set` [command](./commands/argocd_app_set.md) supports more tool-specific flags such as `--kustomize-image`, `--jsonnet-ext-var-str` etc
-flags. You can also specify overrides directly in the source field on application spec. Read more about supported options in corresponded tool [documentation](./application_sources.md).
+The `argocd app set` [command](./commands/argocd_app_set.md) supports more tool-specific flags such as `--kustomize-image`, `--jsonnet-ext-var-str`, etc.
+You can also specify overrides directly in the source field on application spec. Read more about supported options in corresponding tool [documentation](./application_sources.md).
```
```suggestion
You can also specify overrides directly in the source field on the application spec. Read more about supported options in the corresponding tool [documentation](./application_sources.md).
```

rumstead commented on docs/user-guide/projects.md:
```diff
@@ -335,12 +335,12 @@ stringData:
+    that do _not_ have a `project` set).
 
-All the examples above talk about Git repositories, but the same principles apply to clusters as well.
+All the examples above concerns Git repositories, but the same principles apply to clusters as well.
```
```suggestion
All the examples above concern Git repositories, but the same principles apply to clusters as well.
```

rumstead commented:
@olivergondza thanks for the PR! Can you please sign the DCO? I also added some NIT suggestions on the PR to build on your rewording.

olivergondza commented:
> @olivergondza thanks for the PR! Can you please sign the DCO? I also added some NIT suggestions on the PR to build on your rewording.

Fixed. I relied that `commit.gpgsign=true` will do the trick, good catch.

olivergondza commented:
/bns:delete
